include/dlaf/eigensolver/band_to_tridiag/mc.h

```diff
@@ -269,12 +355,70 @@ struct BandBlock {
   }
 #endif
 
+  SizeType memoryIndex(SizeType j) {
+    if (dist) {
+      DLAF_ASSERT_HEAVY(block_size_ * id_ <= j && j < size_, j, id_, block_size_, size_);
+      return (j - block_size_ * id_) % mem_size_col_;
+    }
+    else {
+      DLAF_ASSERT_HEAVY(0 <= j && j < size_, j, size_);
+      return j;
+    }
+  }
+
   SizeType size_;
   SizeType band_size_;
   SizeType ld_;
+
+  SizeType id_;
+  SizeType block_size_;
+  SizeType mem_size_col_;
+
   memory::MemoryView<T, Device::CPU> mem_;
 };
 
+template <class CommSender, class T, class DepSender>
+[[nodiscard]] auto scheduleSendCol(CommSender&& pcomm, comm::IndexT_MPI dest, comm::IndexT_MPI tag,
+                                   SizeType b, std::shared_ptr<BandBlock<T, true>> a_block, SizeType j,
+                                   DepSender&& dep) {
+  using dlaf::comm::internal::transformMPI;
+  using dlaf::internal::whenAllLift;
+
+  namespace ex = pika::execution::experimental;
+
+  auto send = [dest, tag, b, j](const comm::Communicator& comm,
+                                std::shared_ptr<BandBlock<T, true>> a_block, MPI_Request* req) {
+    DLAF_MPI_CHECK_ERROR(MPI_Isend(a_block->ptr(0, j), to_int(2 * b), dlaf::comm::mpi_datatype<T>::type,
+                                   dest, tag, comm, req));
+  };
+
+  auto comm_sender =
+      whenAllLift(std::forward<CommSender>(pcomm), a_block, std::forward<DepSender>(dep)) |
+      transformMPI(send);
+  // Ensure lifetime shared pointer.
+  return whenAllLift(std::move(a_block), std::move(comm_sender)) | ex::then([](auto) {});
```

Comment:
Is this last `whenAllLift(std::move(a_block), ...)` there just to make sure `a_block` lives long enough? If yes, I think you can rewrite it like this since `transformMPI` will ensure that the values live long enough:

```suggestion
  auto send = [dest, tag, b, j](const comm::Communicator& comm,
                                // The ref here is not a must, but avoids an extra copy of the shared_ptr
                                std::shared_ptr<BandBlock<T, true>>& a_block, MPI_Request* req) {
    DLAF_MPI_CHECK_ERROR(MPI_Isend(a_block->ptr(0, j), to_int(2 * b), dlaf::comm::mpi_datatype<T>::type,
                                   dest, tag, comm, req));
  };

  return whenAllLift(std::forward<CommSender>(pcomm), a_block, std::forward<DepSender>(dep)) |
      transformMPI(send);
```
Or is there another reason for the `whenAllLift`?